src/http.ts

```diff
@@ -58,16 +58,25 @@ export interface MWSOptions {
 }
 
 type HttpMethod = 'GET' | 'POST'
-type ParameterTypes = string | number | (number | string)[] | object[] | boolean | undefined
+type ParameterTypes =
+  | string
+  | number
+  | (number | string)[]
+  | object[]
+  | boolean
+  | { [key: string]: ParameterTypes }
```

Comment:
Better expressed as `Record<string, ParameterTypes>`

Reply:
I tried that at first, but it threw `Type alias 'ParameterTypes' circularly references itself.` and only used the above as a fix. Any reason why `{ [key: string]: ParameterTypes }` won't work?

This is where I got the "fix" https://github.com/microsoft/TypeScript/issues/14174. I didn't read through everything though so there may be something I missed in that issue that might've been a better fix